pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -341,15 +366,9 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, errors.Errorf("machine %s doesn't have any ip addresses defined",
 			machine.Name)
 	}
-	ipAddress := ""
-	for _, address := range machine.Status.Addresses {
-		if address.Type == core.NodeInternalIP {
-			ipAddress = address.Address
-		}
-	}
-	if len(ipAddress) == 0 {
-		return reconcile.Result{}, errors.Errorf("no internal ip address associated with machine %s",
-			machine.Name)
+	addresses, _ := r.getNodeAddress(machine)
```

Comment:
Ideally we need a struct like `MachineInfo` that encapsulates all the metadata that we care about the Machine and pass that around.